### PR TITLE
[380] Add User and Provider models

### PR DESCRIPTION
**Pull request by aliuk2012**
### Changes proposed in this pull request
- Adds Provider model to record just the Provider `name`
- Adds User model to record users `first_name`, `last_name`,`email`  
- creates a relationship between these two models so a `User` can have one `Provider` and a Provider could have many `User`

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -82,8 +90,8 @@
     t.text "full_time_part_time"
     t.boolean "teaching_scholars"
     t.uuid "dttp_id"
-    t.integer "record_type"
     t.text "middle_names"
+    t.integer "record_type"
```
Comment (aliuk2012) [inline db/schema.rb:94]: I've noticed for some odd reason this attribute seems to change order for some reason after a db migrate

File: db/schema.rb [hunk 2]
```diff
@@ -71,6 +77,8 @@
     t.date "date_of_birth"
     t.text "ethnicity"
     t.text "disability"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
     t.text "address_line_one"
```
Comment (aliuk2012) [inline db/schema.rb:82]: I've noticed for some odd reason this attribute seems to change order for some reason after a db migrate

**Comment by aliuk2012** [comment]
[[S] Add user and provider models](https://trello.com/c/ab0BMwx7/380-s-add-user-and-provider-models)

**Review by dankmitchell — APPROVED** [review]

**Review by defong — APPROVED** [review]

File: db/migrate/20201106111550_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    create_table :users do |t|
+      t.string :first_name, null: false
+      t.string :last_name, null: false
+      t.string :email, null: false
+      t.references :provider, index: true, null: false, foreign_key: { to_table: :providers }
+
+      t.timestamps
+    end
+
+    add_index :users, %i[email provider_id]
```
Comment (gpeng) [inline db/migrate/20201106111550_create_users.rb:12]: I'm not sure this is right. It's creating a composite index. Is that what you meant? We should index `email`, `provider_id` is already indexed as a result of line 7.

**Review by gpeng — CHANGES_REQUESTED** [review]

File: db/migrate/20201106111550_create_users.rb (hunk 3 above)
Comment (aliuk2012) [reply db/migrate/20201106111550_create_users.rb:12]: Trello ticket says....
> index on email and provider_id

Line 7 produces https://github.com/DFE-Digital/register-trainee-teachers/pull/147/files#diff-cbf8b25d6d853acf58fa9057841f407d29ffe90649c75cf9e3f51b2d6e3aa1d3R120

Comment (gpeng) [reply db/migrate/20201106111550_create_users.rb:12]:
```suggestion
    add_index :users, :email
```

**Review by gpeng — APPROVED** [review]

**Pull request merged**